auslib/web/api_validator.py

```diff
@@ -0,0 +1,24 @@
+import logging
+from connexion.decorators.validation import ParameterValidator
+
+log = logging.getLogger(__name__)
+
+
+class BalrogParameterValidator(ParameterValidator):
+    """
+    This class is used to keep the flask "defaults" feature behavior on spec validation level.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BalrogParameterValidator, self).__init__(*args, **kwargs)
+
+    def validate_path_parameter(self, param, request):
+        # Keep the base class behavior.
+        val = request.path_params.get(param['name'].replace('-', '_'))
+
+        # Supporting the flask add_url_rule "defaults" parameter to the swagger path parameter.
+        if val is None and 'default' in param:
+            val = param['default']
+            request.path_params[param['name']] = val
```

Comment:
Yeah, defaults are supported, but for query parameters, not for path parameters.
See http://swagger.io/specification/#parameter-object-44

required: "...If the parameter is in "path", this property is required and its value MUST be true...."

default: "...Note: "default" has no meaning for required parameters..."

If you remove the validator, the defaults parameters are not included in the path_parameters dict (see line 22).